google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -169,6 +177,8 @@ public PartitionResponse partitionQuery(
   @Override
   public PartitionResponse partitionRead(
       PartitionReadRequest request, @Nullable Map<Option, ?> options) throws SpannerException {
-    throw new UnsupportedOperationException("Not implemented yet.");
+    // TODO(pongad): Figure out metadata
+    // TODO(pongad): Figure out channel affinity
```

Comment:
I'll investigate this.

Reply:
This will basically need enhancing the channel pool in gax to take in an affinity key.

Reply:
This is being done in a linked PR to GAX. Once merged and released, I'll resolve this.